zookeeper-server/src/main/java/org/apache/zookeeper/AsyncCallback.java

```diff
@@ -241,130 +232,110 @@
-         * - the node has children and some API calls cannnot succeed,
-         * e.g. {@link
-         * org.apache.zookeeper.ZooKeeper#delete(String, int,
-         * org.apache.zookeeper.AsyncCallback.VoidCallback, Object)}.
-         * </li>
+         *  <li>{@link KeeperException.Code#NONODE}
+         *             - The node on given path doesn't exist for some API calls.</li>
+         *  <li>{@link KeeperException.Code#BADVERSION}
+         *             - The given version doesn't match the node's version for some API calls.</li>
+         *  <li>{@link KeeperException.Code#NOTEMPTY}
+         *             - the node has children and some API calls cannot succeed, e.g.
+         *             {@link ZooKeeper#delete(String, int, VoidCallback, Object)}.</li>
          * </ul>
          *
          * @param rc   The return code or the result of the call.
          * @param path The path that we passed to asynchronous calls.
-         * @param ctx  Whatever context object that we passed to
-         *             asynchronous calls.
+         * @param ctx  Whatever context object that we passed to asynchronous calls.
+         *
+         * @see ZooKeeper#delete(String, int, VoidCallback, Object)
+         * @see ZooKeeper#removeAllWatches(String, Watcher.WatcherType, boolean, VoidCallback, Object)
+         * @see ZooKeeper#removeWatches(String, Watcher, Watcher.WatcherType, boolean, VoidCallback, Object)
+         * @see ZooKeeper#sync(String, VoidCallback, Object)
+         *
          */
         void processResult(int rc, String path, Object ctx);
 
     }
 
     /**
-     * This callback is used to process the multiple results from
-     * a single multi call.
-     * See {@link org.apache.zookeeper.ZooKeeper#multi} for more information.
+     * This callback is used to process the multiple results from a single multi call.
      */
     @InterfaceAudience.Public
     interface MultiCallback extends AsyncCallback {
 
         /**
          * Process the result of the asynchronous call.
-         * <p>
-         * On success, rc is
-         * {@link org.apache.zookeeper.KeeperException.Code#OK}.
-         * All opResults are
-         * non-{@link org.apache.zookeeper.OpResult.ErrorResult},
-         *
-         * <p>
-         * On failure, rc is a failure code in
-         * {@link org.apache.zookeeper.KeeperException.Code}.
-         * Either opResults is null, or all opResults are
-         * {@link org.apache.zookeeper.OpResult.ErrorResult}.
-         * All operations will be rolled back even if operations
-         * before the failing one were successful.
+         *
+         * <p>On success, rc is {@link KeeperException.Code#OK}. All {@code opResults} are
+         * non-{@link OpResult.ErrorResult}.
+         *
+         * <p>On failure, rc is a failure code in {@link KeeperException.Code}. Either
+         * {@code opResults} is null, or all {@code opResults} are {@link OpResult.ErrorResult}.
```

Comment:
`all {@code opResults} are {@link OpResult.ErrorResult}.` @breed IIRC ops before the failing one were successful OpResult?